dhall/src/Dhall/Import.hs

```diff
@@ -788,12 +788,9 @@ getCacheDirectory = alternative₀ <|> alternative₁
             Just xdgCacheHome -> return xdgCacheHome
             Nothing           -> empty
 
-    alternative₁ = do
-        maybeHomeDirectory <- liftIO (System.Environment.lookupEnv "HOME")
+    alternative₁ = liftIO (fmap (<> ".cache") Directory.getHomeDirectory)
 
-        case maybeHomeDirectory of
-            Just homeDirectory -> return (homeDirectory </> ".cache")
-            Nothing            -> empty
+    alternative₂ = pure ".cache"
```

Comment:
This looks like a departure from [the standard](https://github.com/dhall-lang/dhall-lang/blob/master/standard/imports.md):

> * Store the cached expression in "${XDG_CACHE_HOME}/dhall/1220${base16Hash}" if the $XDG_CACHE_HOME environment variable is defined and the path is readable and writeable
> * Otherwise, store the cached expression in "${HOME}/.cache/dhall/1220${base16Hash}" if the $HOME environment variable is defined and the path is readable and writeable
> * Otherwise, not cache the expression at all

I think it might be worth upstreaming these changes, to make it easier to build Dhall implementations on Windows.

Reply:
Uh, it is very tied to *nix systems, but i guess it should be precise enough to make all impls save in the same locations. The easy change would be to add the windows alternatives: `%XDG_CACHE_HOME%\...` and %USERPROFILE%\.cache...` and the new fallback to relative dir.

Btw, do you think this pr has to wait to upstream changes?

Reply:
> Btw, do you think this pr has to wait to upstream changes?

No. It would just be good to have a standard that also applies to Windows!